agents/structs.py

```diff
@@ -211,6 +211,7 @@ class FrameData(BaseModel):
     action_input: ActionInput = Field(default_factory=lambda: ActionInput())
     guid: Optional[str] = None
     full_reset: bool = False
+    available_actions: list[GameAction] = Field(default_factory=list)
```

Comment:
```suggestion
    available_actions: list[int] = Field(default_factory=list)
```

it should be `int` as there is no conversions happens here?

Reply:
that isn't an accurate reflection of the code though, try this

```
from pydantic import BaseModel,Field
from enum import Enum
from typing import Any, Optional, Type, Union

class SimpleAction(BaseModel):
    game_id: str = ""


class ComplexAction(BaseModel):
    game_id: str = ""
    x: int = Field(default=0, ge=0, le=63)
    y: int = Field(default=0, ge=0, le=63)

class GameAction(Enum):
    RESET = (0, SimpleAction)
    ACTION1 = (1, SimpleAction)
    ACTION2 = (2, SimpleAction)
    ACTION3 = (3, SimpleAction)
    ACTION4 = (4, SimpleAction)
    ACTION5 = (5, SimpleAction)
    ACTION6 = (6, ComplexAction)

    action_type: Union[Type[SimpleAction], Type[ComplexAction]]
    action_data: Union[SimpleAction, ComplexAction]
    reasoning: Optional[Any]

    def __init__(
        self,
        action_id: int,
        action_type: Union[Type[SimpleAction], Type[ComplexAction]],
    ) -> None:
        self._value_ = action_id
        self.action_type = action_type
        self.action_data = action_type()
        self.reasoning = None

class FrameData(BaseModel):
    available_actions: list[GameAction] = Field(default_factory=list)

print(FrameData.model_validate({'available_actions': [1]}))
```

Reply:
`_value_` attribute takes care of it. Thanks 🙏